src/KubernetesClient/Informers/SharedKubernetesInformer.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using k8s.Informers.Cache;
+using k8s.Informers.Notifications;
+using k8s.Models;
+using Microsoft.Extensions.Logging;
+
+namespace k8s.Informers
+{
+    /// <summary>
+    ///     Opens a single connection to API server with per unique <see cref="KubernetesInformerOptions" />
+    ///     and attaches 1 or more internal subscriber to it. The connection is automatically opened if there is
+    ///     at least one subscriber and closes if there are none
+    /// </summary>
+    /// <typeparam name="TResource">The type of resource to monitor</typeparam>
+    public class SharedKubernetesInformer<TResource> :
+        SharedOptionsInformer<TResource, KubernetesInformerOptions>,
```

Comment:
This would benefit from `KubernetesInformerOptions` being made immutable and equatable; currently, a new `SharedInformer<,>` will be created for options that are considered equal.

Reply:
That was always the intent to allow subscribers that have the same options to share the same shared informer (which is locked to a single type of options). I could have swore I implemented equality operators. Thanks for spotting this.